projects/igniteui-angular/src/lib/grids/grid/grid.component.ts

```diff
@@ -743,6 +743,12 @@ export class IgxGridComponent extends IgxGridBaseComponent implements OnInit, Do
         const columnExpr = sortingExpr.find((expr) => expr.fieldName === event.owner.id);
         columnExpr.dir = 3 - columnExpr.dir;
         this.sort(columnExpr);
+        this.onSortingDone.emit({
```

Comment:
The event is emitted from the wrong component, before and after your fix. The event should be emitted by the grid, whenever `sort()` is called.

Currently it's emitted by the `IgxHeaderComponent` on click, or by the `IgxChip` in the group by area on click. Both should be removed and `onSortingDone` should be emitted by the `sort()` method, as just mentioned.